features/dd-sdk-android-flags/src/main/kotlin/com/datadog/android/flags/featureflags/internal/model/DatadogEvaluationContext.kt

```diff
@@ -0,0 +1,124 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.flags.featureflags.internal.model
+
+import com.datadog.android.api.InternalLogger
+import com.datadog.android.flags.featureflags.model.EvaluationContext
+
+/**
+ * Internal representation of evaluation context that contains only normalized
+ * attributes that Datadog's targeting system can parse and use in evaluation rules.
+ *
+ * This class handles the conversion from the flexible public EvaluationContext
+ * to the strictly typed format required by the API (Map<String, String>).
+ *
+ * Key features:
+ * - Validates targeting key is not blank/whitespace-only
+ * - Normalizes all attribute values to String type
+ * - Filters out unsupported types and null values
+ * - Provides validation methods for context integrity
+ *
+ * @param targetingKey The identifier used for bucketing and targeting. Must remain consistent
+ *                     for the same entity to ensure consistent flag evaluation across requests.
+ *                     Must not be blank or whitespace-only.
+ * @param attributes Map of normalized attributes where all values are String representations
+ *                   of the original attribute values from the public EvaluationContext.
+ */
+internal class DatadogEvaluationContext internal constructor(
+    /** The identifier used for bucketing and targeting in flag evaluation. */
+    val targetingKey: String,
+    val attributes: Map<String, String>
+) {
+
+    /**
+     * Validates the Datadog evaluation context.
+     * @return true if the context is valid, false otherwise.
+     */
+    fun isValid(): Boolean = targetingKey.isNotBlank()
+
+    companion object {
+        private const val UNSUPPORTED_VALUE_WARNING =
+            "Unsupported attribute value type for key '%s': %s. " +
+                "Only String, Number, and Boolean values are supported. Attribute skipped."
+        private const val NULL_VALUE_WARNING =
+            "Null attribute value for key '%s'. Attribute skipped."
+
+        /**
+         * Creates a DatadogEvaluationContext from a public EvaluationContext,
+         * normalizing all attributes to String key-value pairs that the API can handle.
```

Comment:
```suggestion
         * Creates a [DatadogEvaluationContext] from a public [EvaluationContext],
         * normalizing all attributes to [String] key-value pairs that the API can handle.
```

It is better to put `[]` around class names to have a link-like navigation when reading docs. See KDoc syntax reference https://kotlinlang.org/docs/kotlin-doc.html.

Same applies to other KDoc entries in the module codebase.